CUDADataFormats/SiPixelCluster/interface/SiPixelClustersCUDA.h

```diff
@@ -27,28 +27,18 @@ class SiPixelClustersCUDA {
   uint32_t *moduleId() { return moduleId_d.get(); }
   uint32_t *clusModuleStart() { return clusModuleStart_d.get(); }
 
-  uint32_t const *moduleStart() const { return moduleStart_d.get(); }
-  uint32_t const *clusInModule() const { return clusInModule_d.get(); }
-  uint32_t const *moduleId() const { return moduleId_d.get(); }
-  uint32_t const *clusModuleStart() const { return clusModuleStart_d.get(); }
```

Comment:
This (and the equivalent in SiPixelDigisCUDA.h) is the only change I'm not very fond of.
I've seen the comment by Slava - but, if `std::vector` can happily have both `begin()` and `cbegin()` *etc.* I don't see why we cannot have the same here.

However, could you remind me why we actually need  `c_moduleStart()` *etc.* ?
What I mean is, if somebody wants a `const*`, why not just write
```c++
uint32_t const* start = clusters.moduleStart();
```
instead of
```
uint32_t const* start = clusters.c_moduleStart();
```
?

Moreover, all use cases I've found are in `SiPixelRawToClusterGPUKernel.cu`, and are used to pass a `const *` argument to a kernel.
Using `.moduleInd()` instead of `.c_moduleInd()` *etc.* should work just fine in this case, and the pointer would still be `const` in the kernel function.

Am I missing something else ?

Reply:
Using non-`const` `clusModuleStart()` would indeed work. The idea of `c_clusModuleStart()` is for a case where the returned pointer is passed directly to a function along `func(clusters.moduleStart(), ...)` and the caller wants to ensure that the function gets a pointer-to-`const`. The `func()` may today take a pointer-to-`const`, but someone could change the function to take a pointer-to-non-`const` and modify the contents of the pointed-to-memory. The `c_clusModuleStart()` approach (or explicit `const_cast`) would help to catch that mismatch with a compilation error.

Written that, I suppose for this particular case the question is more academic, given that all the uses of non-`const` `clusters` is in a single file (plus a few `#include`s) and that this pattern for SoA would better not spread much. So I can change to the other way as well (but would really like to avoid changing again after that :) ).

Reply:
The last three commits show the impact of dropping `c_` prefix. If we go with this approach, I'll squash them into the respective commits having other changes of the affected files.